x/mongo/driver/topology/server.go

```diff
@@ -333,7 +380,9 @@ func (s *Server) ProcessError(err error) {
 		return
 	}
 
-	// updates description to unknown
+	// For a non-timeout network error, we clear the pool, set the description to Unknown, and cancel the in-progress
+	// monitoring check.
+	s.cancelCheck()
```

Comment:
Does the order matter here? Might be safer to cancel the check _last_ to avoid a post-cancellation reconnect attempt racing with this updateDescription call:
```
s.updateDescription(...)
s.pool.clear()
s.cancelCheck()
```